File: sadik-app/src/api/usage.ts
import { http } from './http';

export interface UsageStats {
  period_days: number;
  total_turns: number;
  avg_turns_per_day: number;
  total_audio_seconds: number;
  avg_audio_seconds_per_turn: number;
  p50_total_ms: number;
  p95_total_ms: number;
  avg_stt_ms: number;
  avg_llm_ttfb_ms: number;
  avg_tts_ttfb_ms: number;
  top_tools: Array<{ name: string; count: number }>;
  total_prompt_tokens: number;
  total_completion_tokens: number;
  estimated_cost_usd: number;
  cost_breakdown: { stt_usd: number; llm_usd: number; tts_usd: number };
}

export const usageApi = {
  getMine: (days: number = 30) =>
    http.get<UsageStats>(`/api/usage/me?days=${days}`).then((r) => r.data),
};

File: sadik-backend/app/schemas/usage.py
from pydantic import BaseModel


class UsageStats(BaseModel):
    period_days: int
    total_turns: int
    avg_turns_per_day: float
    total_audio_seconds: float
    avg_audio_seconds_per_turn: float
    p50_total_ms: int
    p95_total_ms: int
    avg_stt_ms: int
    avg_llm_ttfb_ms: int
    top_tools: list[dict]  # [{name, count}]
    total_prompt_tokens: int
    total_completion_tokens: int
    estimated_cost_usd: float
    cost_breakdown: dict  # {stt_usd, llm_usd, tts_usd}
